src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -789,6 +806,18 @@ protected theorem extractLsb_ofNat (x n : Nat) (hi lo : Nat) :
     getLsbD (extractLsb hi lo x) i = (i ≤ (hi-lo) && getLsbD x (lo+i)) := by
   simp [getLsbD, Nat.lt_succ]
 
+@[simp] theorem getMsbD_extractLsb {hi lo : Nat} {x : BitVec w} {i : Nat} :
+    (extractLsb hi lo x).getMsbD i
+    = (decide (i < hi - lo + 1)
+      && (decide (lo + (hi - lo - i) < w)
```

Comment:
Why the strange parenthesisation here? You should just cancel the `lo`s.

Reply:
Similarly in the last `&&`. You can make sure of the condition from the first `decide`.

Reply:
Now updated - changing the statement too with the new condition